scripts/zones/The_Garden_of_RuHmet/mobs/Jailer_of_Faith.lua

```diff
@@ -7,27 +7,114 @@ mixins = { require("scripts/mixins/job_special") }
+    mob:addListener("TAKE_DAMAGE", "FAITH_TAKE_DAMAGE", function(mobArg, amount, attacker, attackType, damageType)
+        if attackType == xi.attackType.PHYSICAL then
+            mobArg:setLocalVar("PhysicalDamage", mobArg:getLocalVar("PhysicalDamage") + amount)
+        elseif attackType == xi.attackType.MAGICAL then
+            mobArg:setLocalVar("MagicalDamage", mobArg:getLocalVar("MagicalDamage") + amount)
+        elseif attackType == xi.attackType.RANGED then
+            mobArg:setLocalVar("RangedDamage", mobArg:getLocalVar("RangedDamage") + amount)
+        elseif attackType == xi.attackType.BREATH then
+            mobArg:setLocalVar("BreathDamage", mobArg:getLocalVar("BreathDamage") + amount)
+        else
+            -- ignore Untyped Damage
+        end
+    end)
+end
+
 entity.onMobSpawn = function(mob)
+    mob:setLocalVar("PhysicalDamage", 0)
+    mob:setLocalVar("MagicalDamage", 0)
+    mob:setLocalVar("RangedDamage", 0)
+    mob:setLocalVar("BreathDamage", 0)
+    xi.mix.jobSpecial.config(mob, {
+        specials =
+        {
+            {
+                id = xi.jsa.MANAFONT,
+                endCode = function(mobArg) -- Uses Manafont and casts Quake II -- only casts this during manafont.
+                    mobArg:castSpell(211) -- quake II
+                end,
+            },
+        },
+    })
     -- Change animation to open
     mob:setAnimationSub(2)
+    mob:setMobMod(xi.mobMod.HP_STANDBACK, 0)
+    mob:setMod(xi.mod.FASTCAST, 30) -- taken from timing of spells from multiple captures
+    mob:setMod(xi.mod.DEF, 450)
 end
 
 entity.onMobFight = function(mob)
-    -- Forms: 0 = Closed  1 = Closed  2 = Open 3 = Closed
-    local randomTime = math.random(45, 180)
-    local changeTime = mob:getLocalVar("changeTime")
-
-    if mob:getBattleTime() - changeTime > randomTime then
-        -- Change close to open.
-        if (mob:getAnimationSub() == 1) then
-            mob:setAnimationSub(2)
-        else -- Change from open to close
-            mob:setAnimationSub(1)
+    mob:addListener("COMBAT_TICK", "FAITH_CTICK", function(mobArg)
+        local sum = mobArg:getLocalVar("PhysicalDamage") + mobArg:getLocalVar("MagicalDamage") + mobArg:getLocalVar("RangedDamage") + mobArg:getLocalVar("BreathDamage")
+        if mobArg:getAnimationSub() == 2 and sum > 1500 then -- Faith will close flower upon taking 1500 damage combined.
+            closeFlower(mobArg)
+        elseif mobArg:getAnimationSub() <= 1 and mobArg:getBattleTime() > mobArg:getLocalVar("[faith]changeTime") then
+            openFlower(mobArg)
+        else
+            -- if no dmg taken - dont trigger a change
         end
-        mob:setLocalVar("changeTime", mob:getBattleTime())
+    end)
+end
+
+entity.onMobMagicPrepare = function(mob, target, spell)
+    local rnd = math.random()
+    if rnd < 0.5 and mob:hasStatusEffect(xi.effect.MANAFONT) == true then -- quake II replaces existing earth damage spells during manafont
```

Comment:
this code doesn't accomplish what the comment says, only has a 50% ish chance of selecting quake, and then will still mov eon to those other earth spells. Either the comment ain't right or the logic ain't right.

Reply:
```
[21:23:53] [Jailer・of・Faith] Manafont → Jailer・of・Faith
[21:23:57] [Jailer・of・Faith] Quake II → [Redacted]
[21:23:57] ([Redacted]) open, was closed
[21:24:04] [Jailer・of・Faith] 60 Quake II → [Redacted]
[21:24:08] [Jailer・of・Faith] Slowga → [Redacted]
[21:24:11] Jailer・of・Faith Slowga → [Redacted] (Resists)
[21:24:12] [[Redacted]] Flash → Jailer・of・Faith
[21:24:13] [Redacted] Flash → Jailer・of・Faith (Flash)
[21:24:14] [Jailer・of・Faith] missed → [Redacted]
[21:24:14] [Jailer・of・Faith] Quake II → [Redacted]
[21:24:19] The Jailer of Faith is no longer afflicted with Flash.
[21:24:22] [Jailer・of・Faith] 60 Quake II → [Redacted]
[21:24:26] [Jailer・of・Faith] Breakga → [Redacted]
[21:24:31] Jailer・of・Faith Breakga → [Redacted] (Resists)
[21:24:34] [Jailer・of・Faith] Stonega III → [Redacted]
[21:24:39] [Jailer・of・Faith] 39 Stonega III → [Redacted]
[21:24:41] [Jailer・of・Faith] missed → [Redacted]
[21:24:42] [Jailer・of・Faith] Quake II → [Redacted]
[21:24:50] [Jailer・of・Faith] 60 Quake II → [Redacted]
[21:24:54] [Jailer・of・Faith] Stone IV → [Redacted]
[21:24:59] [Jailer・of・Faith] 25 Stone IV → [Redacted]
[21:26:23] [Jailer・of・Faith] Manafont → Jailer・of・Faith
[21:26:23] [Jailer・of・Faith] Viscid Nectar → [Redacted]
[21:26:24] [Jailer・of・Faith] Viscid Nectar (Miss) → [Redacted]
[21:26:31] [Jailer・of・Faith] Slowga → [Redacted]
[21:26:33] Jailer・of・Faith Slowga → [Redacted] (Resists)
[21:26:37] [Jailer・of・Faith] Quake II → [Redacted]
[21:26:44] [Jailer・of・Faith] 60 Quake II → [Redacted]
[21:26:47] [2] [Jailer・of・Faith] missed → [Redacted]
[21:26:48] [Jailer・of・Faith] Stone IV → [Redacted]
[21:26:53] [Jailer・of・Faith] 25 Stone IV → [Redacted]
[21:26:53] {2}: Cornelia and Sakura Afterglow effect wears off.
[21:26:57] [Jailer・of・Faith] Stonega III → [Redacted]
[21:27:02] [Jailer・of・Faith] 162 Stonega III → [Redacted]
[21:27:05] [Jailer・of・Faith] Stone IV → [Redacted]
[21:27:10] [Jailer・of・Faith] 25 Stone IV → [Redacted]
[21:27:13] [Jailer・of・Faith] missed → [Redacted]
[21:27:14] [Jailer・of・Faith] Quake II → [Redacted]
[21:27:22] [Jailer・of・Faith] 60 Quake II → [Redacted]
[21:27:24] [Jailer・of・Faith] Manafont → Jailer・of・Faith
[21:27:25] [Jailer・of・Faith] Quake II → [Redacted]
[21:27:33] [Jailer・of・Faith] 60 Quake II → [Redacted]
[21:27:36] [Jailer・of・Faith] Stonega III → [Redacted]
[21:27:41] [Jailer・of・Faith] 39 Stonega III → [Redacted]
[21:27:44] [Jailer・of・Faith] missed → [Redacted]
[21:27:45] [Jailer・of・Faith] Quake II → [Redacted]
[21:27:52] [Jailer・of・Faith] 60 Quake II → [Redacted]
[21:27:56] [Jailer・of・Faith] Stone IV → [Redacted]
[21:28:01] [Jailer・of・Faith] 25 Stone IV → [Redacted]
[21:28:05] [Jailer・of・Faith] Breakga → [Redacted]
[21:28:10] Jailer・of・Faith Breakga → [Redacted] (Resists!)
[21:28:12] [Jailer・of・Faith] missed → [Redacted]
[21:28:13] [Jailer・of・Faith] Stone IV → [Redacted]
[21:28:19] [Jailer・of・Faith] 25 Stone IV → [Redacted]
[21:28:23] [Jailer・of・Faith] Stonega III → [Redacted]
```

Not a whole lot of data, but with 3 manafonts it seems that it doesn't always choose Quake 2, but it might choose Quake 2 immediately after manafont.

Reply:
> doesn't always choose Quake 2

So comment not _exactly_ right then